legal-api/src/legal_api/models/filing.py

```diff
@@ -426,9 +426,7 @@ def set_processed(self):
         """Assign the completion and effective dates, unless they are already set."""
         if not self._completion_date:
             self._completion_date = datetime.utcnow()
-        if (self.effective_date is None or
-            (self.payment_completion_date
-             and self.effective_date < self.payment_completion_date)):  # pylint: disable=W0143; hybrid property
+        if (self.effective_date is None and self.payment_completion_date):  # pylint: disable=W0143; hybrid property
```

Comment:
The front-end only allows a user to select a date for the director change or agm date from the current date or earlier. Thus the effective date would always be less than the payment_completion_date always changing the effective date to the payment date

Reply:
As per our conversation @lambert-alex , if the FE is populating the effective date properly for all the scenarios outlined in the ticket for BEN/COOP COD and AR filings, then I think you only need to update this `set_processed` function.  The update would be to allow an effective date that is before the payment completion date for BEN/COOP COD and AR filings.  The previous logic of using the payment completion date as the effective date would still remain for other applicable scenarios.

And it would be ideal to have effective date validation for annual reports which is currently missing but I think we can do that as another ticket.

@lekshmimallika-aot @vysakh-menon-aot does this sound alright to you?

Reply:
sounds good to me